File: Text_Editor_Project/Core/EditorMemento.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Text_Editor_Project.Core
{
    public class EditorMemento
    {
        public string Content { get; set; }
        public int CursorPosition { get; set; }

        public List<string> Selections { get; set; }

        public EditorMemento(string content, int cursorPos, List<string> selections) {

            Content = content;
            CursorPosition = cursorPos;
            Selections = new List<string>(selections);
        
        
        }
        public EditorMemento()
        {
            Content = string.Empty;
            CursorPosition = 0;
            Selections = new List<string>();
        }

     





    }
}

File: TextEditor/src/EditorMemento.java
import java.util.List;
public class EditorMemento {
    private String content;
    private int cursorPosition;
    private List<String> selections;

    public EditorMemento(String content, int cursorPosition, List<String> selections){
        this.content = content;
        this.cursorPosition = cursorPosition;
        this.selections = selections;

    }

    public String getContent() {
        return content;
    }

    public int getCursorPosition()
    {
        return cursorPosition;
    }

    public List<String> getSelections(){
        return selections;
    }

}
